assets/javascripts/discourse/components/group-assignments-filter.js.es6

```diff
@@ -1,5 +1,15 @@
 import Component from "@ember/component";
 
 export default Component.extend({
-  tagName: "li"
+  tagName: "li",
+  displayName: "",
+
+  init(){
+    if(this.siteSettings.prioritize_username_in_ux){
+      this.set("displayName",this.filter.username);
+    }else{
+      this.set("displayName",this.filter.displayName);
+    }
+    this._super(...arguments);
+  }
```

Comment:
I think a better approach here would be to use a computed property. If you search core for `@discourseComputed` you'll find some examples. [For example](https://github.com/discourse/discourse/blob/fbf29443bdb84e0a07a34a82d7893ab967854bd9/app/assets/javascripts/admin/components/admin-report-table-cell.js#L11-L14).

Reply:
Solved this in new commit

Reply:
This should be computed based on all the properties it uses. So `filter.username` and `filter.displayName` should also be included in the `discourseComputed(...)`

Reply:
Sure, just added these too in discourseComputed() in latest commit

Reply:
And then we can pass them as variables into the function, just like you did for the site setting :)

Reply:
Oooh sorry for that😅😅

Reply:
Updated this one😄